medical/models/medical_abstract_entity.py

```diff
@@ -0,0 +1,207 @@
+                    ('code', '=', category_code),
+                ])
+                self.env['res.partner.id_number'].create({
+                    'partner_id': record.partner_id.id,
+                    'category_id': category.id,
+                    'name': getattr(record, field_name),
+                })
+            elif record_len == 1:
+                value = getattr(record, field_name)
+                id_number.name = value
+            else:
+                raise UserError(_(
+                    'This %s has multiple IDs of this type (%s), so a write '
+                    'via the %s field is not possible. In order to fix this, '
+                    'please use the IDs tab.',
+                ) % (
+                    record._name, category_code.name, field_name,
+                ))
+
+    @api.model
+    def _create_vals(self, vals):
+        """ Override this in child classes in order to add default values. """
+        if self._allow_image_create(vals):
+            vals['image'] = self._get_default_image(vals)
+        return vals
+
+    @api.model_cr_context
+    def _allow_image_create(self, vals):
+        """ It determines if conditions are present that should stop image gen.
+
+        This is implemented so that tests aren't wildly creating images left
+         and right for no reason. Child classes could also inherit this to
+         provide custom rules for image generation.
+
+        Note that this method explicitly allows image generation if
+         ``__image_create_allow`` is a ``True`` value in the context. Any
+         child that chooses to provide custom rules shall also adhere to this
+         context, unless there is a documented reason to not do so.
+        """
+        if vals.get('image'):
+            return False
+        if any((getattr(threading.currentThread(), 'testing', False),
+                self._context.get('install_mode'))):
+            if not self.env.context.get('__image_create_allow'):
+                return False
+        return True
+
+    @api.model_cr_context
+    def _get_default_image(self, vals):
+        """ Overload this in child classes in order to add a default image.
+
+        Child classes should only add the image if super returns False/None.
+        They should return a base64 encoded image.
+
+        Example:
+
+            .. code-block:: python
+
+            @api.model
+            def _get_default_image(self, vals):
+                res = super(MedicalPatient, self)._get_default_image(vals)
+                if not res:
+                    return res
+                img_path = odoo.modules.get_module_resource(
+                    'base', 'static/src/img', 'patient-avatar.png',
+                )
+                with open(img_path, 'r') as image:
+                    base64_image = image.read().encode('base64')
+                    return odoo.tools.image_resize_image_big(base64_image)
```

Comment:
@lasley The source of the problem ;)

`return` should be outside the `with` because we are done with the `image` already.